workflows/diagnostics/fv3net/diagnostics/offline/compute_diagnostics.py

```diff
@@ -242,6 +243,13 @@ def weighted_mean(ds, weights, dims):
         return (ds * weights).sum(dims) / weights.sum(dims)
 
 
+def _maybe_average_over_dataset(ds: xr.Dataset) -> xr.Dataset:
```

Comment:
Instead of each compute function doing the 'maybe average' as a last step, could the average over datasets be done a level up by splitting the test data into the various `dataset` coordinates if that dim is present and calling `compute_diagnostics` on each of these before averaging them together at the end?

e.g. something like this at this part of the code https://github.com/ai2cm/fv3net/blob/00851a37066a1efce949dd3a2a868a4fef650d9e/workflows/diagnostics/fv3net/diagnostics/offline/compute.py#L164

```
if `dataset` in ds.dims:
    for dataset_coord in ds.dataset:
         dataset_summaries.append( compute_diagnostics(
            prediction.sel(dataset=dataset_coord ), target.sel(dataset=dataset_coord ), grid, ds[DELP], n_jobs=n_jobs
        )

... and then avg over the resulting summary datasets
```

Reply:
As discussed on Slack, we need to be careful about computing the r2 values, but it should be possible to preserve per-dataset metrics in the computed diagnostics while continuing to display purely aggregate metrics in the report.  

f6c776d is a start at implementing things this way, but needs to be cleaned up.  I will do that tomorrow.